biorxiv/journal_tracker/map_preprints_to_journals.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Following the Preprint to Published Path
+
+# The goal of this notebook is to map preprint dois to published dois and published dois to Pubmed Central articles.
+
+# In[1]:
+
+
+from habanero import Crossref
+import pandas as pd
+from ratelimit import limits, sleep_and_retry
+import tqdm
+from urllib.error import HTTPError
+
+
+# In[2]:
+
+
+preprints_df = pd.read_csv("../exploratory_data_analysis/output/biorxiv_article_metadata.tsv", sep="\t")
+preprints_df.head()
+
+
+# In[3]:
+
+
+dois = (
+    preprints_df
+    .doi
+    .unique()
+)
+print(len(dois))
+
+
+# In[4]:
+
+
+cf = Crossref(mailto="dnicholson329@gmail.com")
+
+
+# In[5]:
+
+
+TEN_MINUTES = 600
+
+@sleep_and_retry
+@limits(calls=10, period=TEN_MINUTES)
+def call_crossref(doi_ids):
+    responses = []
+    for doi in doi_ids:
+        try:
+            response = cf.works(ids=doi)
+            responses.append(response)
+        except:
+            responses.append({
+                "message":{
+                    "relation":{"none":"none"}, 
+                    "DOI":doi
+                }
+            })
+        
+    return responses
+
+
+# # Map preprint DOIs to Published DOIs
+
+# In[8]:
+
+
+batch_limit = 100
+doi_mapper_records = []
+
+for batch in tqdm.tqdm(range(0, len(dois), batch_limit)):
```

Comment:
> How is this mapping from preprint doi to pub doi working? Is this in the json somewhere?

It works like this:
1. Take a preprint doi and ask cross ref is there a published doi that matches this preprint?
2. If yes capture the published doi version otherwise skip the preprint doi
3. The the next step is to cycle through the published doi and ask the following question
4. What is the journal that maps to this doi? If not found put a place holder and make @danich1  manually fill it in. Only had a couple that I needed to fill in.

The call_crossref makes an api call to the crossref server and the responses variable is a list of json objects that gets returned.